### PR TITLE
Support for BG functions

### PR DESCRIPTION
**Pull request by taten**
Adds support for background functions for GCP functions

## What did you implement:

Closes #XXXXX

## Todos:

_**Note: Run `npm run test-ci` to run all validation checks on proposed changes**_

- [x] Write tests and confirm existing functionality is not broken.
       **Validate via `npm test`**
- [ ] Write documentation
- [x] Ensure there are no lint errors.
       **Validate via `npm run lint-updated`**
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [ ] Ensure introduced changes match Prettier formatting.
       **Validate via `npm run prettier-check-updated`**
       _Note: All reported issues can be automatically fixed by running `npm run prettify-updated`_
- [x] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

**_Is this ready for review?:_** NO
**_Is it a breaking change?:_** NO

### CODE REVIEW

**Comment by FabioOrlinski** [comment]
Background function working evidence:

image
image

Test code coverage:

image

File: gcp/src/gcpContext.ts [hunk 1]
```diff
@@ -17,17 +17,25 @@ export class GcpContext implements CloudContext {
...
 
-    this.runtime = {
-      event: args[0],
-      context: args[1],
-      callback: args[2],
-    };
+    const isBackgroundFunction = args && args[1].eventId;
+    if (isBackgroundFunction) {
+      this.runtime = new GcpBackgroundFunctionRuntime();
+      this.runtime.event = args[0];
```
Comment (eliasscaceres) [inline gcp/src/gcpContext.ts]: code duplication in this if/else method. 
      this.runtime.event = args[0];
      this.runtime.context = args[1];

File: samples/gcp-storage/src/storageHandler.js [hunk 2]
```diff
@@ -20,3 +20,15 @@ module.exports.handler = app.use(
     context.send({ result }, 200);
   }
 );
+
+module.exports.helloGCS = app.use([], async (context) => {
+  const file = context.event;
+  // console.log(`  Event: ${context.context.eventId}`);
+  // console.log(`  Event Type: ${context.eventType}`);
```
Comment (eliasscaceres) [inline samples/gcp-storage/src/storageHandler.js]: avoid commented code.

File: gcp/src/gcpContext.ts (hunk 1 above)
Comment (FabioOrlinski) [reply gcp/src/gcpContext.ts]: Done f4b732fe39b43b9cf4003d35fbe084bad905ab5d

File: samples/gcp-storage/src/storageHandler.js (hunk 2 above)
Comment (FabioOrlinski) [reply samples/gcp-storage/src/storageHandler.js]: Done f4b732fe39b43b9cf4003d35fbe084bad905ab5d

File: samples/gcp-storage/README.md [hunk 3]
```diff
@@ -105,6 +105,10 @@ So, the next step is just create a get request using postman, as the following i
 
 ![postman-response](./images/result-postman.png)
 
+When it create or modify a file in the bucket, it trigger the background function.
```
Comment (taten) [inline samples/gcp-storage/README.md]: Assuming the function doesn't create or modifies the file i'd change this to: 

"When a file gets created or modified in the bucket, the background function gets triggered"

So it is more clear that the function gets triggered when something happens in the bucket

Comment (FabioOrlinski) [reply samples/gcp-storage/README.md]: Done ec70c4127c3d03fe77b57d6f234f1e8ed39ac6b6

**Review by FabioOrlinski — APPROVED** [review]

**Pull request merged**